src/Features/Core/Portable/RemoveUnusedVariable/AbstractRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -154,8 +154,12 @@ private void MergeNodesToRemove(HashSet<SyntaxNode> nodesToRemove)
             var candidateLocalDeclarationsToRemove = new HashSet<TLocalDeclarationStatement>();
             foreach (var variableDeclarator in nodesToRemove.OfType<TVariableDeclarator>())
             {
-                var localDeclaration = (TLocalDeclarationStatement)variableDeclarator.Parent.Parent;
-                candidateLocalDeclarationsToRemove.Add(localDeclaration);
+                //NOTE: we can't cast directly here since in for loops the daclarator.Parent.Parent
+                //is the ForStatementSyntax instead of the TLocalDeclarationStatement
+                if (variableDeclarator.Parent.Parent is TLocalDeclarationStatement localDeclaration)
```

Comment:
wont' this not work for VB?

Reply:
speaking of which, do we need VB tests for this scenario?

Reply:
I don't think it is a problem in VB since vb doesn't allow specifying a `for` with more than one declared variable and said variable is always "used" by the looping logic. Even if it werent (i.e. `For index = 0 To 0`) removing the variable/declaration would result in illegal VB syntax.

Reply:
Also, do you know why the `roslyn-CI` and `roslyn-integration-CI` pipelines are seemingly stuck on `Waiting for status to be reported`? Should i be worried?

Reply:
> I don't think it is a problem in VB since it doesn't allow specifying a for with more than one declared variable and said variable is always "used" by the looping logic. 

So the latter is again requiring that we depend on external behavior that we cannot guarantee.  If things ever change such that the variable is *not* used by teh looping logic, then this will not work.  We shouldn't encode C#isms into shared code.

Reply:
I thought the readson i gave after addressed that. But you are right the syntax might change in the future to possibly allow a for statement without variable declaration.

Sorry if you have to repeat yourself. I think i am scared of breaking something and therefore try to change as little as possible.

Reply:
You have *zero* to be sorry about.  These are all great questions.  And your concerns are totally understandable. :)   Ths is why we have PRs.  So you can share your ideas and thoughts, but also hear the other side of the coin to give you confidence about things you might have been wary about.

Note: if it will help, you can also reach people in realtime at gitter.im/dotnet/roslyn or discord.gg/csharp (#roslyn)

Reply:
I will have to look into the gitter and discord sometime soon :)
I added the requested abstract method and am returning nothing from VB for now which can then be changed should the allowed syntax change.